include/sparrow/arrow_array_schema_proxy.hpp

```diff
@@ -95,100 +132,180 @@ namespace sparrow
         [[nodiscard]] ArrowSchema& schema();
         [[nodiscard]] const ArrowSchema& schema() const;
 
-        [[nodiscard]] std::size_t
-        compute_buffer_size(buffer_type buffer_type, int64_t length, data_type data_type) const;
-    };
+        arrow_schema_private_data* get_schema_private_data();
+        arrow_array_private_data* get_array_private_data();
 
-    inline
-    void arrow_proxy::initialize_children()
-    {
-        m_children.clear();
-        m_children.reserve(static_cast<std::size_t>(array().n_children));
-        for (int64_t i = 0; i < array().n_children; ++i)
-        {
-            m_children.emplace_back(array().children[i], schema().children[i]);
-        }
-    }
+        std::pair<ArrowSchema, ArrowSchema> copy_data() const;
```

Comment:
This declaration can be removed, it is not called nor implemented.

Reply:
done